tests/e2e_test.go

```diff
@@ -0,0 +1,419 @@
+			DestinationChainID: common.Hash(blockchainIDB),
+			DestinationAddress: teleporterContractAddress,
+			Payload:            payload,
+		})
+		gomega.Expect(err).Should(gomega.BeNil())
+
+		// Send a transaction that simulates the Teleporter contract calling sendWarpMessage with a Teleporter message payload
+		tx := types.NewTx(&types.DynamicFeeTx{
+			ChainID:   chainAIDInt,
+			Nonce:     nonceA,
+			To:        &warp.Module.Address,
+			Gas:       200_000,
+			GasFeeCap: big.NewInt(225 * params.GWei),
+			GasTipCap: big.NewInt(params.GWei),
+			Value:     common.Big0,
+			Data:      packedInput,
+		})
+		txSigner := types.LatestSignerForChainID(chainAIDInt)
+		signedTx, err := types.SignTx(tx, txSigner, teleporterKey)
+		gomega.Expect(err).Should(gomega.BeNil())
+
+		// Sleep for some time to make sure relayer has started up and subscribed.
+		time.Sleep(15 * time.Second)
+		log.Info("Subscribing to new heads on destination chain")
+
+		newHeadsB := make(chan *types.Header, 10)
+		sub, err := chainBWSClient.SubscribeNewHead(ctx, newHeadsB)
+		gomega.Expect(err).Should(gomega.BeNil())
+		defer sub.Unsubscribe()
+
+		log.Info("Sending sendWarpMessage transaction", "destinationChainID", blockchainIDB, "txHash", signedTx.Hash())
+		err = chainAWSClient.SendTransaction(ctx, signedTx)
+		gomega.Expect(err).Should(gomega.BeNil())
+
+		log.Info("Waiting for new block confirmation")
+		var blockHash common.Hash
+		newHead := <-newHeadsB
+		log.Info("Received new head", "height", newHead.Number.Uint64())
+		blockHash = newHead.Hash()
+
+		block, err := chainBWSClient.BlockByHash(ctx, blockHash)
+		gomega.Expect(err).Should(gomega.BeNil())
+		log.Info("Got block", "blockHash", blockHash, "blockNumber", block.NumberU64(), "transactions", block.Transactions(), "block", block)
+		accessLists := block.Transactions()[0].AccessList()
+		gomega.Expect(len(accessLists)).Should(gomega.Equal(1))
+		gomega.Expect(accessLists[0].Address).Should(gomega.Equal(warp.Module.Address))
+
+		// Check the transaction storage key has warp message we're expecting
+		storageKeyHashes := accessLists[0].StorageKeys
+		packedPredicate := predicateutils.HashSliceToBytes(storageKeyHashes)
+		predicateBytes, err := predicateutils.UnpackPredicate(packedPredicate)
+		gomega.Expect(err).Should(gomega.BeNil())
+		receivedWarpMessage, err = avalancheWarp.ParseMessage(predicateBytes)
+		gomega.Expect(err).Should(gomega.BeNil())
+
+		// Check that the transaction has successful receipt status
+		txHash := block.Transactions()[0].Hash()
+		receipt, err := chainBWSClient.TransactionReceipt(ctx, txHash)
+		gomega.Expect(err).Should(gomega.BeNil())
+		gomega.Expect(receipt.Status).Should(gomega.Equal(types.ReceiptStatusSuccessful))
+
+		log.Info("Finished sending warp message, closing down output channel")
+
+		// Cancel the command and stop the relayer
+		relayerCancel()
+		_ = cmd.Wait()
+	})
+
+	ginkgo.It("Verify Warp Message", ginkgo.Label("Relay", "VerifyWarp"), func() {
```

Comment:
nit: Can you rename this step? "Verify warp message" sounds like the aggregate signature verification within Avalanchego, which is not what we're checking here.